sklearn/neighbors/_base.py

```diff
@@ -201,31 +201,76 @@ def _check_precomputed(X):
     copied = graph.format != "csr"
     graph = check_array(graph, accept_sparse="csr")
     check_non_negative(graph, whom="precomputed distance matrix.")
+    graph = sort_graph_by_row_values(graph, copy=not copied, warn_when_not_sorted=True)
 
-    if not _is_sorted_by_data(graph):
+    return graph
+
+
+def sort_graph_by_row_values(graph, copy=False, warn_when_not_sorted=True):
+    """Sort a sparse graph such that each row is stored with increasing values.
+
+    .. versionadded:: 1.2
+
+    Parameters
+    ----------
+    graph : sparse matrix of shape (n_samples, n_samples)
+        Distance matrix to other samples, where only non-zero elements are
+        considered neighbors. Matrix is converted to CSR format if not already.
+
+    copy : bool
+        If True, the graph is copied before sorting. If False, the sorting is
+        performed inplace. If graph is not of CSR format, a copy is always
+        returned.
```

Comment:
Looking at this again, I always thought that `copy=False` sometimes returning a copy is strange in other APIs. What do you think about being strict and raising an error when `graph.format != "csr"` and `copy=False`?

Reply:
Sounds good